src/test/integrasjonstester/kotlin/no/nav/familie/ks/sak/api/BehandlingControllerTest.kt

```diff
@@ -0,0 +1,162 @@
+import no.nav.familie.ks.sak.kjerne.arbeidsfordeling.domene.ArbeidsfordelingPåBehandlingRepository
+import no.nav.familie.ks.sak.kjerne.behandling.domene.Behandling
+import no.nav.familie.ks.sak.kjerne.behandling.domene.BehandlingRepository
+import no.nav.familie.ks.sak.kjerne.behandling.domene.BehandlingÅrsak
+import no.nav.familie.ks.sak.kjerne.fagsak.domene.Fagsak
+import no.nav.familie.ks.sak.kjerne.fagsak.domene.FagsakRepository
+import no.nav.familie.ks.sak.kjerne.personident.AktørRepository
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.hamcrest.CoreMatchers.`is` as Is
+
+class BehandlingControllerTest : OppslagSpringRunnerTest() {
+
+    @Autowired
+    private lateinit var behandlingRepository: BehandlingRepository
+
+    @Autowired
+    private lateinit var fagsakRepository: FagsakRepository
+
+    @Autowired
+    private lateinit var arbeidsfordelingPåBehandlingRepository: ArbeidsfordelingPåBehandlingRepository
+
+    @Autowired
+    private lateinit var aktørRepository: AktørRepository
+
+    @MockkBean
+    private lateinit var integrasjonClient: IntegrasjonClient
+
+    private lateinit var fagsak: Fagsak
+    private lateinit var behandling: Behandling
+
+    @BeforeEach
+    fun beforeEach() {
+        fagsak = lagreFagsak()
+        behandling = lagreBehandling(fagsak)
+    }
+
+    val behandlingControllerUrl = "/api/behandlinger"
+
+    @BeforeEach
+    fun setup() {
+        RestAssured.port = port
+    }
+
+    @Test
+    fun `hentBehandling - skal returnere 401 dersom brukeren ikke har token for å hente behandling`() {
+        When {
+            get("$behandlingControllerUrl/401")
+        } Then {
+            statusCode(HttpStatus.UNAUTHORIZED.value())
+        }
+    }
+
+    @Test
+    fun `hentBehandling - skal returnere 403 dersom brukeren ikke har rettigheter til å hente en behandling`() {
+        val token = lokalTestToken(behandlerRolle = BehandlerRolle.UKJENT)
+
+        Given {
+            header("Authorization", "Bearer $token")
+        } When {
+            get("$behandlingControllerUrl/403")
+        } Then {
+            statusCode(HttpStatus.FORBIDDEN.value())
+        }
+    }
+
+    @Test
+    fun `hentBehandling - skal returnere BehandlingResponsDto og 200 OK dersom behandling finnes`() {
+        val token = lokalTestToken(behandlerRolle = BehandlerRolle.BESLUTTER)
+        every { integrasjonClient.sjekkTilgangTilPersoner(any()) } returns Tilgang(true, "test")
+
+        Given {
+            header("Authorization", "Bearer $token")
+        } When {
+            get("$behandlingControllerUrl/${behandling.id}")
+        } Then {
+            statusCode(HttpStatus.OK.value())
+            body("data.behandlingId", Is(behandling.id.toInt()))
+            body("data.årsak", Is("SØKNAD"))
+        }
+    }
```

Comment:
Gir det mening å legge inn en test for scenarioet hvor `behandlingId` ikke finnes? Og validere at respons er 404 med fornuftig feilmelding?

Reply:
> Gir det mening å legge inn en test for scenarioet hvor `behandlingId` ikke finnes? Og validere at respons er 404 med fornuftig feilmelding?

Forsøkte meg på denne, men det ser ut som at vi returnerer 500 for nesten alle feil 😓 
Må sikkert gjennomgå ks-sak koden litt og se på vår håndtering av feil.